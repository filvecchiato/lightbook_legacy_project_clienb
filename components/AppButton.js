import React from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';

TouchableOpacity.defaultProps = { activeOpacity: 0.8 };

const AppButton = ({ onPress, title }) => (
  <TouchableOpacity onPress={onPress}>
    <Text style={styles.appButtonText}>{title}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  screenContainer: {
    justifyContent: 'center',
  },
  appButtonContainer: {
    width: 60,
    height: 20,
    elevation: 8,
    borderRadius: 10,
    marginTop: 40,
    marginRight: 40,
    paddingVertical: 40,
    paddingHorizontal: 40,
  },
  appButtonText: {
    fontSize: 18,
    color: '#fff',
    fontWeight: 'bold',
    alignSelf: 'center',
  },
});

export default AppButton;
