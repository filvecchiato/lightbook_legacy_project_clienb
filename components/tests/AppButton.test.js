import React from 'react';
import renderer from 'react-test-renderer';
import AppButton from '../AppButton';

test('renders correctly', () => {
  const tree = renderer.create(<AppButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
