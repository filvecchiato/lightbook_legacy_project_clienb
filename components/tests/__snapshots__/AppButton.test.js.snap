// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    test
  </Text>
</View>
`;
